### PR TITLE
UCP/RNDV: Implement rendezvous pipeline progress

### PR DESCRIPTION
**Pull request by yosefe**
## Why
Implement pipelined protocol progress - create fragment requests and send them.

### CODE REVIEW

File: src/ucp/rndv/rndv_ppln.c [hunk 1]
```diff
@@ -116,6 +125,95 @@ ucp_proto_rndv_ppln_init(const ucp_proto_init_params_t *init_params)
...
+    ucs_status_t status;
+    ucp_request_t *freq;
+    uint8_t sg_count;
+
+    /* Nested pipeline is prevented during protocol selection */
+    ucs_assert(!(req->flags & UCP_REQUEST_FLAG_RNDV_FRAG));
+
+    req->send.state.completed_size = 0;
+    req->send.rndv.ppln.send_ack   = 0;
+    rpriv                          = req->send.proto_config->priv;
```
Comment (brminich) [inline src/ucp/rndv/rndv_ppln.c:181]: minor: can be init in-place

File: src/ucp/rndv/proto_rndv.inl [hunk 2]
```diff
@@ -133,6 +130,44 @@ ucp_proto_rndv_rkey_destroy(ucp_request_t *req)
...
+    freq = ucp_request_get(worker);
+    if (freq == NULL) {
+        ucs_error("failed to allocated rendezvous send fragment");
+        return UCS_ERR_NO_MEMORY;
+    }
+
+    ucp_trace_req(req, "allocated rndv fragment %p", freq);
+    ucp_request_set_super(freq, req);
+    freq->flags   = UCP_REQUEST_FLAG_RNDV_FRAG;
+    freq->send.ep = req->send.ep;
```
Comment (brminich) [inline src/ucp/rndv/proto_rndv.inl:147]: maybe also fill some `freq` fields, like `req->send.rndv.remote_req_id` or `rkey` here?

Comment (yosefe) [reply src/ucp/rndv/proto_rndv.inl:147]: ucp_proto_rndv_frag_request_alloc() is also going to be used for 3-stage pipeline sender side when allocating fragment for "put" operation, and then these fields will be filled by ucp_proto_rndv_send_start()

File: src/ucp/rndv/rndv_ppln.c (hunk 1 above)
Comment (yosefe) [reply src/ucp/rndv/rndv_ppln.c:181]: then clang-format will not align by "=" properly because this line would be too long

**Review by brminich — APPROVED** [review]

**Pull request merged**